python/knowledges/batch_upload/batch_upload_advanced.py

```
@@ -0,0 +1,656 @@
+            first_page = response.json()
+            
+            # 估算總頁數（如果有 count 欄位）
+            total_count = first_page.get('count', 0)
+            per_page = len(first_page.get('results', []))
+            estimated_pages = (total_count // per_page) + (1 if total_count % per_page > 0 else 0) if per_page > 0 else 1
+            
+            self.logger.info(f"Estimated {total_count} files across {estimated_pages} pages")
+            
+            # 使用 tqdm 顯示分頁進度
+            with logging_redirect_tqdm():
+                with tqdm(total=estimated_pages, desc="Fetching KB pages", unit="pages") as pbar:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)